### PR TITLE
tpm.py: add source elements for external vtpm

### PR DESCRIPTION
**Pull request by yanqzhan**
Add 'source' element with 'type', 'mode', 'path' attributes in tpm xml for external vtpm configuration.

### CODE REVIEW

**Comment by yanqzhan** [comment]
#avocado run --vt-type libvirt --test-runner=runner tpm_device.normal_test.default_model.external.start_vm
JOB ID     : 3c044f18a27ddc5ba2c9a461147ba33d2321ad7d
JOB LOG    : /var/lib/avocado/job-results/job-2023-04-09T09.53-3c044f1/job.log
 (1/1) type_specific.io-github-autotest-libvirt.virtual_devices.tpm_device.normal_test.default_model.external.start_vm: PASS (67.19 s)
RESULTS    : PASS 1 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /var/lib/avocado/job-results/job-2023-04-09T09.53-3c044f1/results.html
JOB TIME   : 68.64 s

#vim /var/lib/avocado/job-results/job-2023-04-09T09.53-3c044f1/job.log
2023-04-09 09:53:42,907 tpm_device       L0806 DEBUG| tpm dev xml to add is:
 <tpm model="tpm-crb"><backend type="external"><source type="unix" mode="connect" path="/var/tmp/guest-swtpm.sock" /></backend></tpm>

**Review by dzhengfy — APPROVED** [review]
LGTM

**Comment by dzhengfy** [comment]
@yanqzhan Do you mind to provide a python session test result for fetch_attrs() and setup_attrs() ? That will be helpful for others to resue this class for new attributes using fetch_attrs() and setup_attrs() method.

**Comment by yanqzhan** [comment]
> 

I can try recently when I have time.

File: virttest/libvirt_xml/devices/tpm.py [hunk 1]
```diff
@@ -84,6 +90,21 @@ def __init__(self, virsh_instance=base.base.virsh):
                                      subclass=self.ActivePCRBanks,
                                      subclass_dargs={
                                          'virsh_instance': virsh_instance})
+            accessors.XMLAttribute(property_name="source_type",
```
Comment (chloerh) [inline virttest/libvirt_xml/devices/tpm.py:93]: Would you consider setting 'source' as XMLElementDict?

**Comment by yanqzhan** [comment]
> @yanqzhan Do you mind to provide a python session test result for fetch_attrs() and setup_attrs() ? That will be helpful for others to resue this class for new attributes using fetch_attrs() and setup_attrs() method.

Tested PASS with following codes in .py file:
        tpm_dev = Tpm()
        source_attrs = {'source_type': 'unix', 'source_mode': 'connect', 'source_path': '/var/tmp/guest-swtpm.sock'}
        tpm_device_attrs = {'tpm_model': "tpm-crb", 'backend': {'backend_type': "external", 'source_type': "unix", 'source_mode': "connect", 'source_path': "/var/tmp/guest-swtpm.sock"}}
        tpm_dev.setup_attrs(**tpm_device_attrs)
        logging.debug("my vtpm XML: %s", tpm_dev)
        vm_xml.add_device(tpm_dev, True)
        logging.debug("my vm xml after adding tpm device is: %s" % vm_xml)
        tpm_device = vm_xml.get_devices('tpm')[0]
        tpm_attributes = tpm_device.fetch_attrs()
        logging.debug("The attributes of tpm device in vm xml:\n%s", tpm_attributes)
        return

Got msgs in log:
# vim /var/lib/avocado/job-results/job-2023-05-05T01.57-9bcd003/job.log
...
2023-05-05 01:57:59,787 tpm_device       L0754 DEBUG| my vtpm XML: <tpm model="tpm-crb"><backend type="external"><source type="unix" mode="connect" path="/var/tmp/guest-swtpm.sock" /></backend></tpm>
...
2023-05-05 01:58:00,071 tpm_device       L0756 DEBUG| my vm xml after adding tpm device is: <domain xmlns:ns0="http://libosinfo.org/xmlns/libvirt/domain/1.0" type="kvm">
  <name>avocado-vt-vm1</name>
...
    </rng><tpm model="tpm-crb"><backend type="external"><source type="unix" mode="connect" path="/var/tmp/guest-swtpm.sock" /></backend></tpm></devices></domain>
2023-05-05 01:58:00,077 tpm_device       L0759 DEBUG| The attributes of tpm device in vm xml:
{'tpm_model': 'tpm-crb', 'backend': {'source_mode': 'connect', 'source_path': '/var/tmp/guest-swtpm.sock', 'backend_type': 'external', 'source_type': 'unix'}}
...

**Comment by yanqzhan** [comment]
Codes updated and attrs tests also PASS：
        tpm_dev = Tpm()
        tpm_device_attrs = {'tpm_model': 'tpm-crb', 'backend': {'backend_type': 'external', 'source': {'type': 'unix', 'mode': 'connect', 'path': '/var/tmp/guest-swtpm.sock'}}}
        tpm_dev.setup_attrs(**tpm_device_attrs)
        logging.debug("my vtpm XML: %s", tpm_dev)
        vm_xml.add_device(tpm_dev, True)
        logging.debug("my vm xml after adding tpm device is: %s" % vm_xml)
        tpm_device = vm_xml.get_devices('tpm')[0]
        tpm_attributes = tpm_device.fetch_attrs()
        logging.debug("The attributes of tpm device in vm xml:\n%s", tpm_attributes)
        return

2023-05-05 03:18:21,265 tpm_device       L0754 DEBUG| my vtpm XML: <tpm model="tpm-crb"><backend type="external"><source type="unix" mode="connect" path="/var/tmp/guest-swtpm.sock" /></backend></tpm>
2023-05-05 03:18:21,557 tpm_device       L0756 DEBUG| my vm xml after adding tpm device is: <domain xmlns:ns0="http://libosinfo.org/xmlns/libvirt/domain/1.0" type="kvm">
  <name>avocado-vt-vm1</name>
...
    </rng><tpm model="tpm-crb"><backend type="external"><source type="unix" mode="connect" path="/var/tmp/guest-swtpm.sock" /></backend></tpm></devices></domain>
2023-05-05 03:18:21,562 tpm_device       L0759 DEBUG| The attributes of tpm device in vm xml:
{'backend': {'source': {'type': 'unix', 'mode': 'connect', 'path': '/var/tmp/guest-swtpm.sock'}, 'backend_type': 'external'}, 'tpm_model': 'tpm-crb'}

File: virttest/libvirt_xml/devices/tpm.py (hunk 1 above)
Comment (yanqzhan) [reply virttest/libvirt_xml/devices/tpm.py]: Done. Thank you!

**Review by chloerh — APPROVED** [review]
Thank you for such detailed test log!

**Pull request merged**